### ISSUE TITLE
NestedScopeFunctions: Nesting functions with `it` variable

### ISSUE THREAD

**Issue by WildOrangutan** [detekt/detekt#4942]
## Expected Behavior of the rule

Add option, to allow nesting scope functions, that have different scope variable name. For example:

```
input?.let { in ->
    output?.let { out ->
        doSomething(in, out)
    }
}
```

Could be configurable with separate threshold maybe.

## Context

Even tough it's a bad practice to nest scope functions, I think that providing different scope variable name is less problematic. It's harder to get confused what scope-variable is called, if they are named differently.

**Comment by dzirbel** [comment]
Is this a case of [`NoNameShadowing`](https://detekt.dev/docs/rules/naming#nonameshadowing)? I don't follow exactly what you're looking for (is the sample code snippet something you want to encourage or discourage?) but I would think that code like

```
input?.let { input -> ... }
```

should be warned for by `NoNameShadowing`.

**Comment by WildOrangutan** [comment]
No, just an extension for a rule, that was recently added #4780

**Comment by marschwar** [comment]
While I think your are right about providing names explicitly make nesting scope function a bit better, I do not think we should add such a configuration option. This would essentially allow users to nest scope functions 5+ levels deep when they only define a name. This would defeat the purpose of this rule entirely.

**Comment by WildOrangutan** [comment]
Well, it's kind of already possible to do this by increasing the threshold.

Hm, maybe we should instead fix threshold for scopes like "apply" to fail at 2 and have only have "let" threshold configurable, but only if scope variable has been renamed.

**Comment by BraisGabin** [comment]
I don't think we want to do that. Nesting scope functions is bad, their own documentation say that.

**Comment by WildOrangutan** [comment]
Documentation just says to avoid overusing them. Imho, what is considered overuse is debatable. It doesn't say you should never nest them.

**Comment by marschwar** [comment]
You are totally right. I still do not know what this flag should accomplish. If you do not rename `it` in the scope, the `NoNameShadowing` rule will flag this. Therefore this rule will assume that the scope variables are properly named. If you believe that nesting scope functions is not that bad, you can increase the allowed depth or disable the rules entirely.

**Comment by BraisGabin** [comment]
From the documentation:

> Avoid nesting scope functions and be careful when chaining them

So I'm closing this, the alternatives noted by @chao2zhang feel on point to me.

**Comment by hudson155** [comment]
+1 to this issue. I'm seeing it with the following code:
```kotlin
override fun delete(id: String): OrganizationModel =
  jdbi.transaction {
    featureService.listByOrganization(id).forEach { featureService.delete(it.id) }
    hostnameService.listByOrganization(id).forEach { hostnameService.delete(it.id) }
    authService.getByOrganization(id)?.let { authService.delete(it.id) }
    return@transaction organizationStore.delete(id)
  }
```

**Comment by BraisGabin** [comment]
Can't you just remove that `return`?

**Comment by BraisGabin** [comment]
`transaction` is not a scoped function. How is this rule reporting that code? Can you create a failing unit test with that code and open a new issue?
